### ISSUE TITLE
git error when doing `bosh prepare deployment`

### ISSUE THREAD

**Issue by holgerkoser** [cloudfoundry-incubator/bosh-workspace#23]
I tried to follow the example in the readme with the cf-warden deployment but I always get the git error:
```
$ bosh prepare deployment

Cloning release 'cf' to satisfy template references
~/.rbenv/versions/2.0.0/lib/ruby/gems/2.0.0/gems/git-1.2.9.1/lib/git/lib.rb:918:in `command': git '--git-dir=~/GitHub/bosh-workspace/.releases/cf/.git' '--work-tree=~/GitHub/bosh-workspace/.releases/cf' log '-30' '--no-color' '--pretty=raw' 'releases/cf-195.yml'  2>&1:fatal: ambiguous argument 'releases/cf-195.yml': unknown revision or path not in the working tree. (Git::GitExecuteError)
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
  from ~/.rbenv/versions/2.0.0/lib/ruby/gems/2.0.0/gems/git-1.2.9.1/lib/git/lib.rb:847:in `command_lines'
  from ~/.rbenv/versions/2.0.0/lib/ruby/gems/2.0.0/gems/git-1.2.9.1/lib/git/lib.rb:145:in `full_log_commits'
  from ~/.rbenv/versions/2.0.0/lib/ruby/gems/2.0.0/gems/git-1.2.9.1/lib/git/log.rb:119:in `run_log'
  from ~/.rbenv/versions/2.0.0/lib/ruby/gems/2.0.0/gems/git-1.2.9.1/lib/git/log.rb:112:in `check_log'
  from ~/.rbenv/versions/2.0.0/lib/ruby/gems/2.0.0/gems/git-1.2.9.1/lib/git/log.rb:89:in `first'
  from ~/.rbenv/versions/2.0.0/lib/ruby/gems/2.0.0/gems/bosh-workspace-0.8.5/lib/bosh/workspace/release.rb:59:in `version_ref'
  from ~/.rbenv/versions/2.0.0/lib/ruby/gems/2.0.0/gems/bosh-workspace-0.8.5/lib/bosh/workspace/release.rb:18:in `update_repo'
  from ~/.rbenv/versions/2.0.0/lib/ruby/gems/2.0.0/gems/bosh-workspace-0.8.5/lib/bosh/cli/commands/prepare.rb:29:in `block in prepare_release_repos'
  from ~/.rbenv/versions/2.0.0/lib/ruby/gems/2.0.0/gems/bosh-workspace-0.8.5/lib/bosh/cli/commands/prepare.rb:27:in `each'
  from ~/.rbenv/versions/2.0.0/lib/ruby/gems/2.0.0/gems/bosh-workspace-0.8.5/lib/bosh/cli/commands/prepare.rb:27:in `prepare_release_repos'
  from ~/.rbenv/versions/2.0.0/lib/ruby/gems/2.0.0/gems/bosh-workspace-0.8.5/lib/bosh/cli/commands/prepare.rb:17:in `prepare'
  from ~/.rbenv/versions/2.0.0/lib/ruby/gems/2.0.0/gems/bosh_cli-1.2811.0/lib/cli/command_handler.rb:57:in `run'
  from ~/.rbenv/versions/2.0.0/lib/ruby/gems/2.0.0/gems/bosh_cli-1.2811.0/lib/cli/runner.rb:56:in `run'
  from ~/.rbenv/versions/2.0.0/lib/ruby/gems/2.0.0/gems/bosh_cli-1.2811.0/lib/cli/runner.rb:16:in `run'
  from ~/.rbenv/versions/2.0.0/lib/ruby/gems/2.0.0/gems/bosh_cli-1.2811.0/bin/bosh:7:in `<top (required)>'
  from ~/.rbenv/versions/2.0.0/bin/bosh:23:in `load'
  from ~/.rbenv/versions/2.0.0/bin/bosh:23:in `<main>'
```
If execute the git command in the shell i get the same error. If I add `--` before the path to the yml file it works.

The following change in [lib/bosh/workspace/release.rb:59](https://github.com/cloudfoundry-incubator/bosh-workspace/blob/master/lib/bosh/workspace/release.rb#L59) for me solved the problem. 
```patch
    def version_ref
-     @repo.log.object(manifest).first
+     @repo.log.path(manifest).first
    end
```

**Comment by rkoster** [comment]
@holgerkoser thanks for reporting this issue. I'm in [the process of switching](https://github.com/cloudfoundry-incubator/bosh-workspace/compare/da022d93e7ec9842f2230f08f99cc8bb21bd8a81...dd66b5c9359b25e989ccefe15abcb328fcef834d) from the [ruby git](https://github.com/schacon/ruby-git) library to [rugged](https://github.com/libgit2/rugged). Cloud you try if you can reproduce your problem when using the [deployment-patch branch](https://github.com/cloudfoundry-incubator/bosh-workspace/tree/deployment-patch)?

**Comment by holgerkoser** [comment]
@rkoster you are right. The problem does NOT occur when I use the deployment-patch branch of bosh-workspace.

Gemfile
```
source "https://rubygems.org"

gem 'bosh-workspace', :git => 'https://github.com/cloudfoundry-incubator/bosh-workspace.git', :branch => 'deployment-patch'
```

**Comment by rkoster** [comment]
@holgerkoser thanks for testing. Will leave this issue open for reference till the deployment-patch branch has been merged into master.

**Comment by drnic** [comment]
This branch also fixed it for me. (needed to remove corrupted `.release` folder first)

**Comment by rkoster** [comment]
This issue was cause by https://github.com/cloudfoundry-incubator/bosh-workspace/commit/586d39594194a02a0a23b64e984c81082a34a367 so reverting to `0.8.3` should also work

**Comment by rkoster** [comment]
Solved in [0.9.0.rc1](https://rubygems.org/gems/bosh-workspace/versions/0.9.0.rc1)
